solidity/test/Polymer7683.t.sol

```diff
@@ -0,0 +1,392 @@
+        bytes32[] memory orderIds = new bytes32[](1);
+        orderIds[0] = orderId;
+    
+        bytes memory proof = _createRefundProof(destChainId, destContract, orderIds);
+    
+        console2.log("First attempt...");
+        polymer7683.handleRefundWithProof(orderId, proof);
+    
+        console2.log("\nSecond attempt (should process but do nothing)...");
+        polymer7683.handleRefundWithProof(orderId, proof);
+    }
+
+    function test_handleSettlementWithProof_wrongChainId() public {
+        vm.prank(owner);
+        polymer7683.setDestinationContract(destChainId, destContract);
+        
+        bytes32 orderId = bytes32("orderId1");
+        bytes memory fillerData = abi.encode(bytes32("filler1"));
+        
+        // Create proof for wrong chain ID
+        bytes memory proof = _createSettlementProof(999, destContract, orderId, fillerData);
+        
+        vm.expectRevert(Polymer7683.UnregisteredDestinationChain.selector);
+        polymer7683.handleSettlementWithProof(proof);
+    }
+
+    function test_handleSettlementWithProof_unregisteredDestination() public {
+        bytes32 orderId = bytes32("orderId1");
+        bytes memory fillerData = abi.encode(bytes32("filler1"));
+        
+        bytes memory proof = _createSettlementProof(destChainId, destContract, orderId, fillerData);
+        
+        vm.expectRevert(Polymer7683.UnregisteredDestinationChain.selector);
+        polymer7683.handleSettlementWithProof(proof);
+    }
+
+    function test_handleSettlementWithProof_invalidOrderId() public {
+        vm.prank(owner);
+        polymer7683.setDestinationContract(destChainId, destContract);
+        
+        // Create proof for one order ID
+        bytes32 proofOrderId = bytes32("order1");
+        bytes memory fillerData = abi.encode(bytes32("filler1"));
+
+        console2.log("\nTest setup:");
+        console2.log("Proof Order ID:");
+        console2.logBytes32(proofOrderId);
+        console2.log("Submitted Order ID:");
+        console2.logBytes32(bytes32("order2"));
+        
+        bytes memory proof = _createSettlementProof(destChainId, destContract, proofOrderId, fillerData);
+
+        (, , bytes memory topics, bytes memory eventData) = prover.validateEvent(proof);
+
+        console2.log("\nDecoding event data...");
+        (
+            bytes32 decodedOrderId,
+            bytes memory decodedOriginData,
+            bytes memory decodedFillerData
+        ) = abi.decode(eventData, (bytes32, bytes, bytes));
+        console2.log("Decoded Order ID from proof:");
+        console2.logBytes32(decodedOrderId);
+        console2.log("Decoded Origin Data length:", decodedOriginData.length);
+        console2.log("Decoded Filler Data length:", decodedFillerData.length);
+      
+        // Try to submit different order ID
+        bytes32 submittedOrderId = bytes32("order2");
+        
+        vm.expectRevert(Polymer7683.InvalidEventData.selector);
```

Comment:
Is this test reverting?

Reply:
no, it was testing the old logic. I just removed the test since the validation should be done and tested in the base contract.